File: book/snippets/js/src/keys.ts
import {Keys, loadWasmSync} from "@rust-nostr/nostr-sdk"

// ANCHOR: generate
function generate() {
    let keys = Keys.generate();

    console.log("Public key (hex): ", keys.publicKey.toHex());
    console.log("Secret key (hex): ", keys.secretKey.toHex());

    console.log("Public key (bech32): ", keys.publicKey.toBech32());
    console.log("Secret key (bech32): ", keys.secretKey.toBech32());
}
// ANCHOR_END: generate

// ANCHOR: restore
function restore() {
    let keys = Keys.parse("nsec1ufnus6pju578ste3v90xd5m2decpuzpql2295m3sknqcjzyys9ls0qlc85");
    console.log("Secret key (hex): ", keys.secretKey.toHex());
}
// ANCHOR_END: restore

// Load WASM
loadWasmSync();

// Run
generate();
restore();

File: book/snippets/kotlin/shared/src/main/kotlin/rust/nostr/snippets/Keys.kt
package rust.nostr.snippets

import rust.nostr.sdk.*

// ANCHOR: generate
fun generate() {
    val keys = Keys.generate()

    val publicKey = keys.publicKey()
    val secretKey = keys.secretKey()

    println("Public key (hex): ${publicKey.toHex()}")
    println("Secret key (hex): ${secretKey.toHex()}")

    println("Public key (bech32): ${publicKey.toBech32()}")
    println("Secret key (bech32): ${secretKey.toBech32()}")
}
// ANCHOR_END: generate

// ANCHOR: restore
fun restore() {
    // Parse secret key
    var keys = Keys.parse("nsec1j4c6269y9w0q2er2xjw8sv2ehyrtfxq3jwgdlxj6qfn8z4gjsq5qfvfk99")

    // Parse from hex
    var secretKey = SecretKey.parse("6b911fd37cdf5c81d4c0adb1ab7fa822ed253ab0ad9aa18d77257c88b29b718e")
    keys = Keys(secretKey = secretKey)
}
// ANCHOR_END: restore
